starport/pkg/gomodulepath/gomodulepath.go

```diff
@@ -49,12 +51,28 @@ func Parse(rawpath string) (Path, error) {
 	return p, nil
 }
 
+// Find search the Go module in the current and parent paths until finding it.
+func Find(path string) (parsed Path, appPath string, err error) {
+	for len(path) != 0 && path != "." && path != "/" {
+		parsed, err = ParseAt(path)
+		if errors.Is(err, gomodule.ErrGoModNotFound) {
+			path = filepath.Dir(path)
+			continue
+		}
+		return parsed, path, err
+	}
+	return Path{}, "", errors.Wrap(gomodule.ErrGoModNotFound, "could not locate your app's root dir")
+}
+
 // ParseAt parses Go module path of an app resides at path.
 func ParseAt(path string) (Path, error) {
 	parsed, err := gomodule.ParseAt(path)
 	if err != nil {
 		return Path{}, err
 	}
+	if err := cosmosanalysis.ValidateGoMod(parsed); err != nil {
```

Comment:
gomodulepath is a generic package, it cannot know about anything specific to cosmos.